Sources/MapboxMaps/Foundation/Camera/CameraAnimator.swift

```diff
@@ -6,7 +6,7 @@ public class CameraAnimator: NSObject {
     // MARK: Stored Properties
 
     /// Instance of the property animator that will run animations.
-    private let propertyAnimator: UIViewPropertyAnimator
+    private var propertyAnimator: UIViewPropertyAnimator?
```

Comment:
Why is this now an optional var?

This class as a whole would be useless without the property animator and so I think making it optional makes no sense

Reply:
From the PR description:
> Makes the propertyAnimator held by CameraAnimator optional -- and makes it so that a deinit of a CameraAnimator
results in the propertyAnimator being nil'ed out.

Reply:
so when it was a `let` before, the deinit of `CameraAnimator` doesn't automatically nil the `propertyAnimator`. 

What if we make property animator a `weak let` ?

Reply:
Then it will fall out of scope immediately.

Reply:
> so when it was a let before, the deinit of CameraAnimator doesn't automatically nil the propertyAnimator.

Not in all cases.

Reply:
Could we be specific in the case then? I see the solution and why it works, but I'm concerned that someone can technically make a `CameraAnimator` with no property animator and that feels wrong. I don't think we should allow that